remote/process_test.go

```diff
@@ -17,392 +14,37 @@ import (
-				exitCode, err := proc.Wait(ctx)
-				require.Error(t, err)
-				assert.Equal(t, 1, exitCode)
-			},
-		},
-		{
-			Name: "WaitGivesProperExitCodeOnSignalDeath",
-			Case: func(ctx context.Context, t *testing.T, opts *options.Create, makep jasper.ProcessConstructor) {
-				proc, err := makep(ctx, testutil.SleepCreateOpts(100))
-				require.NoError(t, err)
-				require.NotNil(t, proc)
-				sig := syscall.SIGTERM
-				require.NoError(t, proc.Signal(ctx, sig))
-				exitCode, err := proc.Wait(ctx)
-				require.Error(t, err)
-				if runtime.GOOS == "windows" {
-					assert.Equal(t, 1, exitCode)
-				} else {
-					assert.Equal(t, int(sig), exitCode)
-				}
-			},
-		},
-		{
-			Name: "WaitGivesNegativeOneOnAlternativeError",
-			Case: func(ctx context.Context, t *testing.T, opts *options.Create, makep jasper.ProcessConstructor) {
-				proc, err := makep(ctx, testutil.SleepCreateOpts(100))
-				require.NoError(t, err)
-				require.NotNil(t, proc)
-
-				var exitCode int
-				waitFinished := make(chan bool)
-				cctx, cancel := context.WithCancel(ctx)
-				cancel()
-				go func() {
-					exitCode, err = proc.Wait(cctx)
-					waitFinished <- true
-				}()
-				select {
-				case <-waitFinished:
-					require.Error(t, err)
-					assert.Equal(t, -1, exitCode)
-				case <-ctx.Done():
-					assert.Fail(t, "call to Wait() took too long to finish")
-				}
-			},
-		},
-		{
-			Name: "InfoHasTimeoutWhenProcessTimesOut",
-			Case: func(ctx context.Context, t *testing.T, _ *options.Create, makep jasper.ProcessConstructor) {
-				opts := testutil.SleepCreateOpts(100)
-				opts.Timeout = time.Second
-				opts.TimeoutSecs = 1
-				proc, err := makep(ctx, opts)
-				require.NoError(t, err)
-
-				exitCode, err := proc.Wait(ctx)
-				assert.Error(t, err)
-				if runtime.GOOS == "windows" {
-					assert.Equal(t, 1, exitCode)
-				} else {
-					assert.Equal(t, int(syscall.SIGKILL), exitCode)
-				}
-				info := proc.Info(ctx)
-				assert.True(t, info.Timeout)
-			},
-		},
-		{
-			Name: "CallingSignalOnDeadProcessDoesError",
-			Case: func(ctx context.Context, t *testing.T, opts *options.Create, makep jasper.ProcessConstructor) {
-				proc, err := makep(ctx, opts)
-				require.NoError(t, err)
+func TestProcessImplementations(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 
-				_, err = proc.Wait(ctx)
-				assert.NoError(t, err)
+	httpClient := testutil.GetHTTPClient()
+	defer testutil.PutHTTPClient(httpClient)
 
-				err = proc.Signal(ctx, syscall.SIGTERM)
-				require.Error(t, err)
-				assert.True(t, strings.Contains(err.Error(), "cannot signal a process that has terminated"))
-			},
-		},
+	testCases := append(jasper.ProcessTests(), []jasper.ProcessTestCase{
 		{
-			Name: "CompleteAlwaysReturnsTrueWhenProcessIsComplete",
-			Case: func(ctx context.Context, t *testing.T, opts *options.Create, makep jasper.ProcessConstructor) {
-				proc, err := makep(ctx, opts)
+			Name: "RegisterTriggerFails",
```

Comment:
why does it fail?

Reply:
Registering an in-memory function on a remote process is intentionally not supported because it's not serializable.